pytng/pytng.pyx

```diff
@@ -209,22 +234,30 @@ cdef class TNGFile:
             self.reached_eof = True
             raise StopIteration("Reached EOF in read")
 
-        cdef np.ndarray[ndim=2, dtype=np.float32_t, mode='c'] xyz = np.empty((self.n_atoms, 3), dtype=np.float32)
-        cdef float* positions = NULL
-        cdef int64_t stride_length, ok, i, n_values_per_frame
-
-        try:
-            ok = tng_util_pos_read_range(self._traj, self.step, self.step, &positions, &stride_length)
-            if ok != TNG_SUCCESS:
-                raise IOError("error reading frame")
+        cdef MemoryWrapper wrap
+        cdef float* positions
+        wrap = MemoryWrapper(3 * self.n_atoms * sizeof(float))
+        positions = <float*> wrap.ptr
 
-            for i in range(self._n_atoms):
-                for j in range(3):
-                    xyz[i, j] = positions[i*3 + j]
-            xyz *= self.distance_scale
-        finally:
-            if positions != NULL:
-                free(positions)
+        cdef int64_t stride_length, ok, i, n_values_per_frame
+        ok = tng_util_pos_read_range(self._traj, self.step, self.step, &positions, &stride_length)
```

Comment:
thinking about it, why do we need a c pointer here? Can we not just use a typed memory view?

Reply:
Also are you sure `positions` is not reallocated by the API? Since positions was originally a NULL pointer I'm sure the library does allocation, not sure about deallocation.

Reply:
Yeah the C code is doing a `realloc` call (probably do deal with varying number of particles?).  So before we'd pass in a NULL and it'd take care of it for us.  I guess we could remove the `malloc` in our code, but I like the symmetry?

If you use a memoryview / allocate in numpy, you get `python(28077,0x113f355c0) malloc: *** error for object 0xc000000000000000: poin
ter being realloc'd was not allocated` because you did a python allocation which doesn't go via malloc (directly at least).